src/App.tsx

```diff
@@ -1,8 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
+import * as api from './api';
+
+const App = () => {
+  const [posts, setPosts] = useState([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
would be nice to create custom hook  `useFetch`
https://scotch.io/tutorials/create-a-custom-usefetch-react-hook
https://www.npmjs.com/package/use-http

it may  look like over engineering for this task so up to you